### PR TITLE
Add data synch, add offline mode.

### PR DESCRIPTION
**Pull request by eschlange**
* Added config token flow back into app.
* Added offline mode by entering 'offline' as config token.
* Removed saving of session slides with just confirmation buttons.
* Stripped new 'help' value on exercises before persisting.
* Updated settings spec by removing window mock.

[#104735420]

### CODE REVIEW

File: config [hunk 1]
```diff
@@ -10,4 +10,9 @@
     <content src="index.html" />
     <access origin="ACCESS_ORIGIN" />
     <icon src="res/icon.png" />
+
+    <access origin="*" />
+    <allow-navigation href="*" />
+    <allow-intent href="http://*/*" />
```
Comment (ericcf) [inline config]: Does one of the plugins use Intents?

File: src/js/controllers/ConfigurationController.js [hunk 2]
```diff
@@ -19,6 +20,11 @@
           .catch(function() {
             self.alert = 'unable to configure application';
           });
+
+      if(this.configurationToken === 'offline') {
```
Comment (ericcf) [inline src/js/controllers/ConfigurationController.js]: Where does `this.configurationToken` get set to "offline"?

Comment (eschlange) [inline src/js/controllers/ConfigurationController.js]: This occurs when a user inputs the string "offline" into the configuration field.

File: config (hunk 1 above)
Comment (eschlange) [inline config]: I'm checking on this now. It may not be necessary.

File: config [hunk 3]
```diff
@@ -10,4 +10,9 @@
     <content src="index.html" />
     <access origin="ACCESS_ORIGIN" />
     <icon src="res/icon.png" />
+
+    <access origin="*" />
```
Comment (ericcf) [inline config]: If it still works with *.northwestern.edu, I'd like to see that. Otherwise seems ok.

Comment (eschlange) [inline config]: I removed the navigation tag, it was unnecessary. I was unable to get either the *.northwestern.edu origin or the '*://*.northwestern.edu/*' to work.

**Comment by ericcf** [comment]
:leaves:

**Pull request merged**